2025-10-16 16:49:17,003 : INFO : root : Loading config.
2025-10-16 16:49:17,007 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2025-10-16 16:49:17,008 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\f7776651e259183bd84a8c8d686bf868.mpf_cache
2025-10-16 16:49:17,084 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\70233839fc0b792ce11a62f3c18c1950.mpf_cache
2025-10-16 16:49:17,085 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\9c8dea79ca4d15ee886340e992e8cb9d.mpf_cache
2025-10-16 16:49:17,086 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\richa\ttp7\modes\base\config\base.yaml
2025-10-16 16:49:17,086 : INFO : ConfigProcessor : Loading config from file C:\Users\richa\ttp7\modes\base\config\base.yaml.
2025-10-16 16:49:17,100 : INFO : ConfigProcessor : Loading config: C:\Users\richa\ttp7\modes\base\config\base.yaml
2025-10-16 16:49:17,101 : INFO : ConfigProcessor : Config file cache created: C:\Users\richa\AppData\Local\Temp\67a19b095f918111543419ac0271bfb1.mpf_cache
2025-10-16 16:49:17,102 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\be387b0e8ed00556ec738d83cd6947d1.mpf_cache
2025-10-16 16:49:17,102 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\5a8efabd9666d8c71fa755fc28a9bf1e.mpf_cache
2025-10-16 16:49:17,103 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\fb58745e58584e219ff53db445e8e9f2.mpf_cache
2025-10-16 16:49:17,104 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\3a5badf48d988f9fc0f9b093a934917d.mpf_cache
2025-10-16 16:49:17,105 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\9e788d9ac28f3464b985b20edf80a10e.mpf_cache
2025-10-16 16:49:17,106 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\7e426d8ffa907371bf962f3181d8bdf2.mpf_cache
2025-10-16 16:49:17,107 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\b710dc01437f1dfd05f3b5275f67493b.mpf_cache
2025-10-16 16:49:17,108 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\0a23590f4081187a3ecee75c7657616e.mpf_cache
2025-10-16 16:49:17,109 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\374a5edaf0766cb9005f368df852c09d.mpf_cache
2025-10-16 16:49:17,110 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\b50d1c72289d6f8f00cba34766fbb2e5.mpf_cache
2025-10-16 16:49:17,110 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\bfc680b70531e398658191c2a179bce0.mpf_cache
2025-10-16 16:49:17,112 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\069a726c379c7a3dc006889318eb30f2.mpf_cache
2025-10-16 16:49:17,112 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\6159dd3e8b071f5aa615e717784124e9.mpf_cache
2025-10-16 16:49:17,114 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\e382c4765158d2be92c1673fc03a4ad6.mpf_cache
2025-10-16 16:49:17,114 : INFO : ConfigProcessor : Loading config from cache: C:\Users\richa\AppData\Local\Temp\868b3b6361e3aaa0a56bd50d9388247e.mpf_cache
2025-10-16 16:49:17,115 : INFO : Machine : Mission Pinball Framework Core Engine v0.57.0.dev39
2025-10-16 16:49:17,115 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'jsonlogging': False, 'logfile': 'logs\\2025-10-16-16-49-17-mpf-DESKTOP-RK7BE5M.log', 'pause': False, 'production': False, 'text_ui': True, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': None, 'syslog_address': None, 'mc_file_name': None, 'no_sound': False}
2025-10-16 16:49:17,115 : INFO : Machine : MPF path: C:\Users\richa\mpfvirtual4\lib\site-packages\mpf
2025-10-16 16:49:17,115 : INFO : Machine : Machine path: C:\Users\richa\ttp7
2025-10-16 16:49:17,115 : INFO : Machine : Platform: win32
2025-10-16 16:49:17,115 : INFO : Machine : Python executable location: C:\Users\richa\mpfvirtual4\Scripts\python.exe
2025-10-16 16:49:17,115 : INFO : Machine : Python version: 3.9.13 (64-bit)
2025-10-16 16:49:17,117 : INFO : Machine : Initializing MPF...
2025-10-16 16:49:17,513 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_credit_units'====== Args={'value': 12, 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_free_play'====== Args={'value': False, 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_player1_score'====== Args={'value': 0, 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.57.0.dev39', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.57.0.dev39, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,573 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.9.13', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,582 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-10-10.0.19045-SP0', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,582 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,582 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '10', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,582 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.19045', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,582 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2025-10-16 16:49:17,582 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2025-10-16 16:49:17,625 : INFO : EventManager : Event: ======'drop_target_bank_TIME_up'====== Args={}
2025-10-16 16:49:17,626 : INFO : EventManager : Event: ======'drop_target_bank_TRAVEL_up'====== Args={}
2025-10-16 16:49:17,626 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2025-10-16 16:49:17,627 : INFO : SwitchController : Found active switch: ('2', <Platform.SmartVirtual>) <switch.s_trough1>
2025-10-16 16:49:17,627 : INFO : SwitchController : Found active switch: ('3', <Platform.SmartVirtual>) <switch.s_trough2>
2025-10-16 16:49:17,627 : INFO : SwitchController : Found active switch: ('4', <Platform.SmartVirtual>) <switch.s_trough3>
2025-10-16 16:49:17,627 : INFO : SwitchController : Found active switch: ('5', <Platform.SmartVirtual>) <switch.s_trough4>
2025-10-16 16:49:17,627 : INFO : SwitchController : Found active switch: ('6', <Platform.SmartVirtual>) <switch.s_trough5>
2025-10-16 16:49:17,627 : INFO : SwitchController : Found active switch: ('7', <Platform.SmartVirtual>) <switch.s_trough6>
2025-10-16 16:49:17,633 : ERROR : ConfigValidator : Your config contains a value for the setting "widgets:text:widgets:text:fot_size", but this is not a valid setting name.
2025-10-16 16:49:17,634 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...tting name.')>)
handle: <Handle EventManager._queue_task_done(<Task finishe...tting name.')>)>
Traceback (most recent call last):
  File "C:\Users\richa\AppData\Local\Programs\Python\Python39\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\events.py", line 791, in _queue_task_done
    future.result()
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\events.py", line 704, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 64, in _validate_config_item
    device_settings['widgets'] = self.process_widgets(
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 92, in process_widgets
    widget_list.append(self.process_widget(widget))
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 111, in process_widget
    self.machine.config_validator.validate_config('widgets:{}'.format(
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 177, in _validate_config
    self.check_for_invalid_sections(this_spec, source, validation_failure_info)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 336, in check_for_invalid_sections
    raise ConfigFileError('Your config contains a value for the '
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mode_base_started:slides with settings {'widgets': [{'type': 'text', 'text': '(score)', 'number_grouping': True, 'min_digits': 2, 'font_size': 10, '_default_settings': ['type', 'text', 'number_grouping', 'min_digits', 'font_size'], 'x': None, 'y': None, 'anchor_x': 'center', 'anchor_y': 'center', 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': 'PLAYER (number)', 'y': '10', 'x': '10', 'font_size': 5, 'anchor_x': 'left', 'anchor_y': 'bottom', '_default_settings': ['type', 'text', 'y', 'x', 'font_size', 'anchor_x', 'anchor_y'], 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'number_grouping': False, 'min_digits': 0, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': 'BALL (ball)', 'y': '5', 'x': 'right-10', 'anchor_x': 'right', 'anchor_y': 'bottom', 'font_size': 5, '_default_settings': ['type', 'text', 'y', 'x', 'anchor_x', 'anchor_y', 'font_size'], 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'number_grouping': False, 'min_digits': 0, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': '(machine|credits_string)', 'fot_size': 10, 'y': '10%', '_default_settings': ['type', 'text', 'fot_size', 'y']}]} >> Your config contains a value for the setting "widgets:text:widgets:text:fot_size", but this is not a valid setting name. Error Code: CFE-ConfigValidator-2 (https://missionpinball.org/logs)
2025-10-16 16:49:17,696 : INFO : Machine : Starting the main run loop.
2025-10-16 16:49:17,702 : INFO : Machine : Shutting down...
2025-10-16 16:49:17,703 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2025-10-16 16:49:18,698 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\machine.py", line 788, in _run_loop
    raise self._exception['exception']
  File "C:\Users\richa\AppData\Local\Programs\Python\Python39\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\events.py", line 791, in _queue_task_done
    future.result()
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\events.py", line 704, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 64, in _validate_config_item
    device_settings['widgets'] = self.process_widgets(
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 92, in process_widgets
    widget_list.append(self.process_widget(widget))
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 111, in process_widget
    self.machine.config_validator.validate_config('widgets:{}'.format(
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 177, in _validate_config
    self.check_for_invalid_sections(this_spec, source, validation_failure_info)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 336, in check_for_invalid_sections
    raise ConfigFileError('Your config contains a value for the '
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mode_base_started:slides with settings {'widgets': [{'type': 'text', 'text': '(score)', 'number_grouping': True, 'min_digits': 2, 'font_size': 10, '_default_settings': ['type', 'text', 'number_grouping', 'min_digits', 'font_size'], 'x': None, 'y': None, 'anchor_x': 'center', 'anchor_y': 'center', 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': 'PLAYER (number)', 'y': '10', 'x': '10', 'font_size': 5, 'anchor_x': 'left', 'anchor_y': 'bottom', '_default_settings': ['type', 'text', 'y', 'x', 'font_size', 'anchor_x', 'anchor_y'], 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'number_grouping': False, 'min_digits': 0, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': 'BALL (ball)', 'y': '5', 'x': 'right-10', 'anchor_x': 'right', 'anchor_y': 'bottom', 'font_size': 5, '_default_settings': ['type', 'text', 'y', 'x', 'anchor_x', 'anchor_y', 'font_size'], 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'number_grouping': False, 'min_digits': 0, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': '(machine|credits_string)', 'fot_size': 10, 'y': '10%', '_default_settings': ['type', 'text', 'fot_size', 'y']}]} >> Your config contains a value for the setting "widgets:text:widgets:text:fot_size", but this is not a valid setting name. Error Code: CFE-ConfigValidator-2 (https://missionpinball.org/logs)
2025-10-16 16:49:21,123 : ERROR : root : 
Traceback (most recent call last):
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\machine.py", line 788, in _run_loop
    raise self._exception['exception']
  File "C:\Users\richa\AppData\Local\Programs\Python\Python39\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\events.py", line 791, in _queue_task_done
    future.result()
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\events.py", line 704, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 64, in _validate_config_item
    device_settings['widgets'] = self.process_widgets(
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 92, in process_widgets
    widget_list.append(self.process_widget(widget))
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpfmc\config_players\plugins\slide_player.py", line 111, in process_widget
    self.machine.config_validator.validate_config('widgets:{}'.format(
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 177, in _validate_config
    self.check_for_invalid_sections(this_spec, source, validation_failure_info)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\config_validator.py", line 336, in check_for_invalid_sections
    raise ConfigFileError('Your config contains a value for the '
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mode_base_started:slides with settings {'widgets': [{'type': 'text', 'text': '(score)', 'number_grouping': True, 'min_digits': 2, 'font_size': 10, '_default_settings': ['type', 'text', 'number_grouping', 'min_digits', 'font_size'], 'x': None, 'y': None, 'anchor_x': 'center', 'anchor_y': 'center', 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': 'PLAYER (number)', 'y': '10', 'x': '10', 'font_size': 5, 'anchor_x': 'left', 'anchor_y': 'bottom', '_default_settings': ['type', 'text', 'y', 'x', 'font_size', 'anchor_x', 'anchor_y'], 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'number_grouping': False, 'min_digits': 0, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': 'BALL (ball)', 'y': '5', 'x': 'right-10', 'anchor_x': 'right', 'anchor_y': 'bottom', 'font_size': 5, '_default_settings': ['type', 'text', 'y', 'x', 'anchor_x', 'anchor_y', 'font_size'], 'round_anchor_x': None, 'round_anchor_y': None, 'opacity': 1.0, 'z': 0, 'reset_animations_events': {}, 'color': [1.0, 1.0, 1.0, 1.0], 'style': [], 'adjust_top': 0, 'adjust_bottom': 0, 'adjust_left': 0, 'adjust_right': 0, 'expire': None, 'key': None, 'widget': None, 'events_when_added': [], 'events_when_removed': [], 'bitmap_font': False, 'font_kerning': True, 'bold': False, 'italic': False, 'number_grouping': False, 'min_digits': 0, 'halign': 'center', 'valign': 'middle', 'rotation': 0, 'scale': 1.0, 'casing': None, 'outline_color': [1.0, 1.0, 1.0, 1.0], 'outline_width': 0, 'animations': None}, {'type': 'text', 'text': '(machine|credits_string)', 'fot_size': 10, 'y': '10%', '_default_settings': ['type', 'text', 'fot_size', 'y']}]} >> Your config contains a value for the setting "widgets:text:widgets:text:fot_size", but this is not a valid setting name. Error Code: CFE-ConfigValidator-2 (https://missionpinball.org/logs)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\commands\game.py", line 236, in __init__
    self.machine.run()
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\machine.py", line 684, in run
    self._run_loop()
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\machine.py", line 791, in _run_loop
    report_crash(self._exception['exception'], "exception_during_run", self.config)
  File "C:\Users\richa\mpfvirtual4\lib\site-packages\mpf\core\crash_reporter.py", line 236, in report_crash
    response = input("Do you want to report the crash to the MPF team (yes/no/show)? ")
EOFError
